test/test_subverb.py

```diff
@@ -0,0 +1,167 @@
+    result1 = extension.main(context=context1)
+    assert result1 is None
+
+    # Second add (duplicate)
+    context2 = CommandContext(command_name='colcon', args=Mock())
+    context2.args.name = MIXIN_IDENTIFIER
+    context2.args.url = TEST_URL
+    result2 = extension.main(context=context2)
+    assert result2 == (
+        f"A repository with the name '{MIXIN_IDENTIFIER}' already exists"
+    )
+
+
+def test_add_repository_invalid_url():
+    """Test adding a repository with an invalid URL format."""
+    with pytest.raises(ArgumentTypeError, match="must contain '://'"):
+        _url_string('invalid-url')
+
+
+def test_add_repository_invalid_name():
+    """Test adding a repository with an invalid name."""
+    # Forward slash test
+    with pytest.raises(ArgumentTypeError, match="must not contain '/'"):
+        _non_empty_string_without_pathsep(f'{MIXIN_IDENTIFIER}/invalid')
+
+    # Backslash test
+    with pytest.raises(ArgumentTypeError, match="must not contain '\\\\'"):
+        _non_empty_string_without_pathsep(f'{MIXIN_IDENTIFIER}\\invalid')
+
+
+def test_update_mixin(capsys):
+    """Test updating a mixin repository."""
+    add_extension = AddMixinSubverb()
+    add_context = CommandContext(command_name='colcon', args=Mock())
+    add_context.args.name = MIXIN_IDENTIFIER
+    add_context.args.url = TEST_URL
+    add_extension.main(context=add_context)
+
+    update_extension = UpdateMixinSubverb()
+    update_context = CommandContext(command_name='colcon', args=Mock())
+    update_context.args.name = MIXIN_IDENTIFIER
+
+    result = update_extension.main(context=update_context)
+    captured = capsys.readouterr()
+
+    assert result == 0
+    assert f'fetching {MIXIN_IDENTIFIER}:' in captured.out
+    assert 'build-type.mixin' in captured.out
+    assert 'coverage.mixin' in captured.out
+
+
+def test_show_mixins(capsys):
+    """Test showing available mixins against expected files."""
+    add_extension = AddMixinSubverb()
+    add_context = CommandContext(command_name='colcon', args=Mock())
+    add_context.args.name = MIXIN_IDENTIFIER
+    add_context.args.url = TEST_URL
+    add_extension.main(context=add_context)
+
+    update_extension = UpdateMixinSubverb()
+    update_context = CommandContext(command_name='colcon', args=Mock())
+    update_context.args.name = MIXIN_IDENTIFIER
+    update_extension.main(context=update_context)
+
+    # Clear previous output from add/update
+    capsys.readouterr()
+
+    show_extension = ShowMixinSubverb()
+    show_extension.add_arguments(parser=Mock())  # Initialize mixins_by_verb
+    show_context = CommandContext(command_name='colcon', args=Mock())
+
+    # Show all verbs and all mixins
+    show_context.args.verb = None
+    show_context.args.mixin_name = None
+
+    result = show_extension.main(context=show_context)
+    captured = capsys.readouterr()
+
+    assert result is None
+
+    expected_demo_file = Path(__file__).parent.joinpath('expected_mixins', 'demo.txt')
```

Comment:
A more common way to combine path atoms, which also happens to be compact enough to fit this line under the 80 column limit.

```suggestion
    expected_demo_file = Path(__file__).parent / 'expected_mixins' / 'demo.txt'
```

Reply:
Merging this to fix CI